ethtx/tx_sender.go

```diff
@@ -0,0 +1,179 @@
+}
+
+func NewTxSender(
+	db tmDB.DB,
+	keyDir string,
+	from gethCommon.Address,
+	ledgerAddress gethCommon.Address,
+) (*TxSender, error) {
+	logger, err := zap.NewDevelopment()
+	if err != nil {
+		return nil, err
+	}
+
+	rpcURL := viper.GetString(common.FlagEthTxRPCURL)
+	parsedRPCURL, err := url.Parse(rpcURL)
+	if err != nil {
+		return nil, err
+	}
+	secondaryRPCURLStrs := viper.GetStringSlice(common.FlagEthTxSecondaryRPCURLs)
+	parsedSecondaryRPCURLs := make([]*url.URL, len(secondaryRPCURLStrs))
+	for i, urlStr := range secondaryRPCURLStrs {
+		parsed, parseErr := url.Parse(urlStr)
+		if parseErr != nil {
+			return nil, parseErr
+		}
+		parsedSecondaryRPCURLs[i] = parsed
+	}
+	config := &esTypes.Config{
+		Logger:           esLogger.NewZapLogger(logger.Sugar()),
+		BlockTime:        time.Duration(viper.GetInt64(common.FlagEthTxBlockTime)) * time.Second,
+		RPCURL:           parsedRPCURL,
+		SecondaryRPCURLs: parsedSecondaryRPCURLs,
+		ChainID:          big.NewInt(viper.GetInt64(common.FlagEthChainID)),
+
+		HeadTrackerHistoryDepth:  viper.GetInt64(common.FlagEthTxHeadTrackerHistoryDepth),
+		HeadTrackerMaxBufferSize: viper.GetInt(common.FlagEthTxHeadTrackerMaxBufferSize),
+		FinalityDepth:            viper.GetInt64(common.FlagEthTxHeadTrackerFinalityDepth),
+
+		DBPollInterval: time.Duration(viper.GetInt64(common.FlagEthTxDBPollInterval)) * time.Second,
+
+		DefaultGasPrice: big.NewInt(viper.GetInt64(common.FlagEthTxDefaultGasPrice)),
+		MaxGasPrice:     big.NewInt(viper.GetInt64(common.FlagEthTxMaxGasPrice)),
+		GasBumpPercent:  viper.GetUint64(common.FlagEthTxGasBumpPercent),
+		GasBumpWei:      big.NewInt(viper.GetInt64(common.FlagEthTxGasBumpWei)),
+
+		GasBumpThreshold: viper.GetInt64(common.FlagEthTxGasBumpThreshold),
+		GasBumpTxDepth:   viper.GetInt(common.FlagEthTxGasBumpTxDepth),
+	}
+	ethClient, err := esClient.NewImpl(config)
+	if err != nil {
+		return nil, err
+	}
+	err = ethClient.Dial(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	txManagerDB := tmDB.NewPrefixDB(db, []byte(prefixTxManager))
+	store := esStoreTM.NewTMStore(txManagerDB)
+	keyStore := esClient.NewKeyStore(keyDir, esClient.ScryptParams{N: keystore.StandardScryptN, P: keystore.StandardScryptP})
+	err = keyStore.Unlock(viper.GetString(common.FlagEthPassphrase))
+	if err != nil {
+		return nil, err
+	}
+
+	txManager, err := esTxManager.NewTxManager(ethClient, store, keyStore, config)
+	if err != nil {
+		return nil, err
+	}
+	err = txManager.RegisterAccount(from)
+	if err != nil {
+		return nil, err
+	}
+	ledgerABI, err := abi.JSON(strings.NewReader(mainchain.CelerLedgerABI))
+	if err != nil {
+		return nil, err
+	}
+	ledgerInfo := &ContractInfo{
+		ABI:     ledgerABI,
+		Address: ledgerAddress,
+	}
+	return &TxSender{
+		ethClient:  ethClient,
+		from:       from,
+		txManager:  txManager,
+		ledgerInfo: ledgerInfo,
+	}, nil
+}
+
+func (s *TxSender) Start() error {
+	return s.txManager.Start()
+}
+
+func (s *TxSender) AddTx(
```

Comment:
if this func is not to be used in places other than inside `sendTx`, make it part of `sendTx`

Reply:
Done.